src/trezor-keyring.ts

```diff
@@ -281,44 +351,62 @@ class TrezorKeyring extends EventEmitter {
 
   /**
    *
-   * @template {TypedTransaction | OldEthJsTransaction} Transaction
-   * @param {string} address - Hex string address.
-   * @param {number} chainId - Chain ID
-   * @param {Transaction} tx - Instance of either new-style or old-style ethereumjs transaction.
-   * @param {(import('trezor-connect').EthereumSignedTx) => Transaction} handleSigning - Converts signed transaction
+   * @param address - Hex string address.
+   * @param chainId - Chain ID
+   * @param tx - Instance of either new-style or old-style ethereumjs transaction.
+   * @param handleSigning - Converts signed transaction
    * to the same new-style or old-style ethereumjs-tx.
-   * @returns {Promise<Transaction>} The signed transaction, an instance of either new-style or old-style
+   * @returns The signed transaction, an instance of either new-style or old-style
    * ethereumjs transaction.
    */
-  async _signTransaction(address, chainId, tx, handleSigning) {
-    let transaction;
+  async #signTransaction(
+    address: string,
+    chainId: number,
+    tx: TypedTransaction,
+    handleSigning: (tx: EthereumSignedTx) => TypedTransaction,
+  ): Promise<TypedTransaction>;
+
+  async #signTransaction(
+    address: string,
+    chainId: number,
+    tx: OldEthJsTransaction,
+    handleSigning: (tx: EthereumSignedTx) => OldEthJsTransaction,
+  ): Promise<OldEthJsTransaction>;
+
+  async #signTransaction(
+    address: string,
+    chainId: number,
+    tx: any,
+    handleSigning: (tx: EthereumSignedTx) => any,
+  ): Promise<any> {
+    let transaction: EthereumTransaction | EthereumTransactionEIP1559;
     if (isOldStyleEthereumjsTx(tx)) {
       // legacy transaction from ethereumjs-tx package has no .toJSON() function,
       // so we need to convert to hex-strings manually manually
       transaction = {
-        to: this._normalize(tx.to),
-        value: this._normalize(tx.value),
-        data: this._normalize(tx.data),
+        to: this.#normalize(tx.to),
+        value: this.#normalize(tx.value),
+        data: this.#normalize(tx.data),
         chainId,
-        nonce: this._normalize(tx.nonce),
-        gasLimit: this._normalize(tx.gasLimit),
-        gasPrice: this._normalize(tx.gasPrice),
+        nonce: this.#normalize(tx.nonce),
+        gasLimit: this.#normalize(tx.gasLimit),
+        gasPrice: this.#normalize(tx.gasPrice),
       };
     } else {
       // new-style transaction from @ethereumjs/tx package
       // we can just copy tx.toJSON() for everything except chainId, which must be a number
       transaction = {
         ...tx.toJSON(),
         chainId,
-        to: this._normalize(tx.to),
-      };
+        to: this.#normalize(ethUtil.toBuffer(tx.to)),
+      } as EthereumTransaction | EthereumTransactionEIP1559;
```

Comment:
I just checked, maybe my TS engine is playing up, but it looks like we can remove `as EthereumTransaction | EthereumTransactionEIP1559` here too